### DISCUSSION TITLE
map meta key

### DISCUSSION THREAD

**Discussion by TerryQuiet** [JetBrains/ideavim#648] [Q&A]
I have this option in Idea `keymap.windows.as.meta=true`, this allows me to use meta for shortcuts.
I wonder if it possible to map it in ideavim?
![2023-06-05_16-24](https://github.com/JetBrains/ideavim/assets/12786477/9c144631-b0e7-4d9c-85a8-d606dbcaac48)
Like this maybe?
`nnoremap <m-q> :NERDTreeFind <CR>`

**Comment by AlexPl292** [comment]
Well, it doesn't work out of the box, right?

**Reply by TerryQuiet** [reply]
it's not, 
I've tried this 
```
map <m-r> <Action>(RecentFiles)
map <M-r> <Action>(RecentFiles)
```
While `<Action>(RecentFiles)` I can map directly in Idea so it would not be a problem. 
I don't know if this is difficult to implement, but having one more free button for mapping in IdeaVim would be nice xD

**Reply by AlexPl292** [reply]
Okay, I've created an issue to track: https://youtrack.jetbrains.com/issue/VIM-2972/Support-meta-key-with-keymap.windows.as.meta-option

**Comment by citizenmatt** [comment]
Can you try the `<D-{char}>` prefix, please? Tl;DR: IntelliJ's "meta" is not the same as Vim's "meta". 

AIUI, Vim supported the Meta key like Emacs did, and this key became the Alt key over time as keyboards changed. So, Vim supports both `<M-` and `<A-` for the Alt key and calls this "meta" in the docs (see [`:help map-meta-keys`](https://vimhelp.org/map.txt.html#%3Amap-meta-keys)). IdeaVim also treats both `<M-` and `<A-` as Alt modifiers.

IntelliJ's "meta" comes from the JVM/Swing, and it's not clear what the history is, but it definitely treats "meta" and "alt" as different. It seems to use "meta" to describe the Super key, another modifier that's become the CMD key on Mac, and the Windows or de-branded Windows/"Super" key on modern keyboards. The character prefix for this is `<D-`, also supported by IdeaVim.
